### PR TITLE
Import prefix support for dartdoc

### LINKED ISSUE #1402: [new ui.Gradient.linear] doesn't linkify
There's a couple of places where we refer to otherwise-ambiguous terms via the import prefix, e.g. `[ui.Image.width]`, `[new ui.Gradient.linear]`.

### PR DESCRIPTION
**Pull request by jcollins-g**
Fixes #1402.

Implementation of prefix support for dartdoc.  Also includes a restructuring of how constructors are handled.  Class objects and constructors are now explicitly converted back and forth as a result of context information rather than indirectly through the code structure.  Some more refactoring to straighten out edge cases related to this; a total of 70 or so broken links fixed for Flutter as a result.

### CODE REVIEW

File: lib/src/model.dart [hunk 1]
```diff
@@ -1227,7 +1236,10 @@ class Constructor extends ModelElement
   }
 
   @override
-  String get fullyQualifiedName => '${library.name}.$name';
+  String get fullyQualifiedName {
```
Comment (pq) [inline lib/src/model.dart]: consider a ternary and the short form?

```dart
String get fullyQualifiedName => isDefaultConstructor ? super.fullyQualifiedName : '${library.name}.$name';
```

File: test/dartdoc_test.dart [hunk 2]
```diff
@@ -252,7 +252,7 @@ void main() {
       Package p = packageGraph.defaultPackage;
       expect(p.name, 'test_package');
       expect(p.hasDocumentationFile, isTrue);
-      expect(packageGraph.defaultPackage.publicLibraries, hasLength(10));
+      expect(packageGraph.defaultPackage.publicLibraries, hasLength(12));
```
Comment (pq) [inline test/dartdoc_test.dart]: To the casual reader, this magic number looks scary.  Maybe a comment to reassure the unfamiliar?

File: testing/test_package/lib/fake.dart [hunk 3]
```diff
@@ -49,8 +49,11 @@ library fake;
 import 'dart:async';
 import 'dart:collection';
 import 'package:meta/meta.dart' show Required;
-import 'css.dart' as css;
+import 'csspub.dart' as css;
+import 'csspub.dart' as renamedLib2;
```
Comment (pq) [inline testing/test_package/lib/fake.dart]: Oh.  Now this is really pathological.

File: lib/src/model.dart (hunk 1 above)
Comment (jcollins-g) [reply lib/src/model.dart]: ternary makes this line a bit long for me.

File: test/dartdoc_test.dart (hunk 2 above)
Comment (jcollins-g) [reply test/dartdoc_test.dart]: added

File: testing/test_package/lib/fake.dart (hunk 3 above)
Comment (jcollins-g) [reply testing/test_package/lib/fake.dart]: It's test code... that's it's job ;-)

**Pull request merged**